source/common/ssl/context_config_impl.cc

```diff
@@ -77,7 +78,42 @@ ClientContextConfigImpl::ClientContextConfigImpl(const Json::Object& config)
 ServerContextConfigImpl::ServerContextConfigImpl(const envoy::api::v2::DownstreamTlsContext& config)
     : ContextConfigImpl(config.common_tls_context()),
       require_client_certificate_(
-          PROTOBUF_GET_WRAPPED_OR_DEFAULT(config, require_client_certificate, false)) {
+          PROTOBUF_GET_WRAPPED_OR_DEFAULT(config, require_client_certificate, false)),
+      session_ticket_keys_([&config] {
+        std::vector<SessionTicketKey> ret;
+
+        switch (config.session_ticket_keys_type_case()) {
+        case envoy::api::v2::DownstreamTlsContext::kSessionTicketKeys:
+          for (const auto& datasource : config.session_ticket_keys().keys()) {
+            switch (datasource.specifier_case()) {
+            case envoy::api::v2::DataSource::kFilename: {
+              const std::string key_data = Filesystem::fileReadToEnd(datasource.filename());
+              validateAndAppendKey(ret, key_data);
+              break;
+            }
+            case envoy::api::v2::DataSource::kInline: {
+              const std::string& key_data = datasource.inline_();
```

Comment:
FWIW, I suspect this will need to be `const std::string`, because of Google's protobuf string != std::string issue, but I can fix it during next import if that's the case.

Reply:
The generated code at least currently returns a const std::string& .  Are you saying it's going to change in a newer version of protobuf?  If so, I'd say let's fix it when it changes.

Reply:
No, what I meant is that Google's internal version of protobuf uses `string` type that's not `std::string` and we've been accommodating that in Envoy via various hacks, which result in a copy vs reference.

But thinking about that, I've just realized that there are no tests that verify this code path - could you add one successful test that loads key data inline? Thanks!

Reply:
Ok, I understand now.  I'll change it to make it easier on you.  Also adding test coverage.

Reply:
I imported this change locally and it doesn't look that this is issue after all (but that might because of lack of tests), so let's leave it as-is for now. Thank you, though!

Reply:
Too late; I just pushed the fix.  I think the new version is fine, but let me know if you want me to put it back.